api/broker/broker_controller.go

```diff
@@ -150,12 +150,12 @@ func (c *Controller) patchBroker(r *web.Request) (*web.Response, error) {
 		return nil, err
 	}
 
-	if err := transformBrokerCredentials(ctx, broker, c.Encrypter.Encrypt); err != nil {
+	catalog, err := c.getBrokerCatalog(ctx, broker)
```

Comment:
would it make sense to add a test in the `test` folder  /semi-integration ?

There should be already a fake broker setup present there and also there is util for `				common.VerifyBrokerCatalogEndpointInvoked(brokerServer, 1)`

The test should just do a patch with new credentials and verify broker catalog is fetched successfully